plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotSessionProperties.java

```diff
@@ -37,13 +37,19 @@ public class PinotSessionProperties
     private static final String SEGMENTS_PER_SPLIT = "segments_per_split";
     private static final String AGGREGATION_PUSHDOWN_ENABLED = "aggregation_pushdown_enabled";
     private static final String COUNT_DISTINCT_PUSHDOWN_ENABLED = "count_distinct_pushdown_enabled";
+    public static final String GRPC_QUERY_ENFORCE_METADATA_EXCEPTION = "grpc_query_enforce_metadata_exception";
 
     private final List<PropertyMetadata<?>> sessionProperties;
 
     @Inject
     public PinotSessionProperties(PinotConfig pinotConfig)
     {
         sessionProperties = ImmutableList.of(
+                booleanProperty(
+                        GRPC_QUERY_ENFORCE_METADATA_EXCEPTION,
+                        "When true, enforce metadata exception in gRPC query even when response type is metadata",
+                        pinotConfig.isGrpcQueryEnforceMetadataException(),
+                        false),
```

Comment:
Move under COUNT_DISTINCT_PUSHDOWN_ENABLED for consistency with the field order.